### PR TITLE
Provide Support For Accessing Router Schemas

### PR DESCRIPTION
**Pull request by THEjacob1000**
# Added Proper Schemas to the `_metadata` proeprty

## Description

This PR introduces the ability to access proper metadata from the router, the intention is for this to be in future used to broadcast the rpc types and how to use it if the user so chooses. Currently the only way to use the rpc safely is within the same repository as it and I'm trying to broaden it up to allow the ability to be installed if the user so wishes to create a public API.

## Type of Change

- [x] `feat`: New feature (non-breaking change which adds functionality)
- [ ] `fix`: Bug fix (non-breaking change which fixes an issue)
- [ ] `docs`: Documentation changes
- [ ] `style`: Code style changes (formatting, etc.)
- [ ] `refactor`: Code refactoring (no functional changes)
- [ ] `test`: Adding or updating tests
- [ ] `chore`: Maintenance tasks
- [ ] `breaking`: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Affected Packages

- [x] `@jsandy/rpc`
- [ ] `@jsandy/builder`
- [ ] `@jsandy/typescript-config`
- [ ] Root workspace/tooling

## Testing

Please describe how your changes have been tested:

- [x] Unit tests pass (`pnpm test`)
- [x] Type checking passes (`pnpm check-types`)
- [x] Linting passes (`pnpm lint`)
- [x] Build succeeds (`pnpm build`)
- [ ] Manual testing performed

### Test Coverage

- [x] New features include tests
- [ ] Tests cover edge cases
- [x] All tests are passing

## Checklist

- [ ] My code follows the project's style guidelines
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: packages/rpc/src/router.ts [hunk 1]
```diff
@@ -128,27 +128,26 @@ interface WebSocketBindings {
...
-		E
-	>,
-> = Promise<TRouter> | TRouter;
+interface JSONSchema {
+	type?: string | string[];
+	properties?: Record<string, JSONSchema>;
+	required?: string[];
+	additionalProperties?: boolean | JSONSchema;
+	$schema?: string;
+}
```
[the remark this answers is not in the document]
Comment (THEjacob1000) [reply packages/rpc/src/router.ts:137]: I don't have the json-schema package installed in this project

File: packages/rpc/src/__tests__/__mocks__/procedures.mock.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+// Mutation with complex logic
+export const deleteUser = j.procedure
+	.use(authMiddleware)
+	.input(GetUserSchema)
+	.mutation(({ input, ctx }) => {
+		if (input.id === ctx.user.id) {
+			throw new Error("Cannot delete your own account");
+		}
+		return Response.json({ success: true, deletedId: input.id });
+	});
```
[the remark this answers is not in the document]
Comment (THEjacob1000) [reply packages/rpc/src/__tests__/__mocks__/procedures.mock.ts:85]: status 200 is automatic with this router logic

File: packages/rpc/src/router.ts [hunk 3]
```diff
@@ -248,8 +259,27 @@ export class Router<
...
+				};
+			} else if (operation.schema) {
+				// Handle regular operations with single schema
+				schema = toJSONSchema(operation.schema);
+			}
+
 			this._metadata.procedures[path] = {
 				type: operation.type,
+				schema,
 			};
```
[the remark this answers is not in the document]
Comment (THEjacob1000) [reply packages/rpc/src/router.ts:283]: the `type` parameter is already inside operation.schema or operation.incoming/outgoing

File: packages/rpc/src/__tests__/schema.test.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+	combinedRouter,
+	userRouter,
+} from "./__mocks__/router.mock";
+
+vi.mock("hono/client", () => ({
+	hc: vi.fn(() => ({
+		$get: vi.fn(),
+		$post: vi.fn(),
+	})),
+}));
```
[the remark this answers is not in the document]
Comment (THEjacob1000) [reply packages/rpc/src/__tests__/schema.test.ts:16]: our hono client on supports get and post

File: packages/rpc/src/__tests__/schema.test.ts [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+describe("Schema", () => {
+	beforeEach(() => {
+		vi.clearAllMocks();
+		global.fetch = vi.fn();
+	});
```
[the remark this answers is not in the document]
Comment (THEjacob1000) [reply packages/rpc/src/__tests__/schema.test.ts]: Removed entirely as we override the fetch in the test setup so this was an unnecessary overwrite
